python/tank/authentication/login_dialog.py

```diff
@@ -17,13 +17,22 @@
 at any point.
 --------------------------------------------------------------------------------
 """
+import socket
 
-from .ui import resources_rc
 from .ui import login_dialog
 from . import session_cache
+from shotgun_shared import saml2_sso
 from .errors import AuthenticationError
 from .ui.qt_abstraction import QtGui, QtCore
-from tank_vendor.shotgun_api3 import MissingTwoFactorAuthenticationFault
+from tank_vendor.shotgun_api3 import Shotgun, MissingTwoFactorAuthenticationFault
+from tank_vendor.shotgun_api3.lib.httplib2 import ServerNotFoundError
+from tank_vendor.shotgun_api3.lib.xmlrpclib import ProtocolError
+from .. import LogManager
+
+logger = LogManager.get_logger(__name__)
+
+# Name used to identify the client application when connecting via SSO to Shotugn.
+TOOLKIT_PRODUCT_NAME = "toolkit"
```

Comment:
This is not specific to SSO... RV (event before I started to work on SSO) used 'rv' as product identifier when creating Shotgun sessions.

So I will go simply with PRODUCT_IDENTIFIER